src/gen9_avc_encoder.c

```diff
@@ -0,0 +1,7615 @@
+        "mb code buffer");
+    if (!allocate_flag)
+        goto failed_allocation;
+
+    size = frame_mb_nums * 32 * 4;
+    allocate_flag = i965_allocate_gpe_resource(i965->intel.bufmgr,
+        &avc_surface->res_mv_data_surface,
+        ALIGN(size,0x1000),
+        "mv data buffer");
+    if (!allocate_flag)
+        goto failed_allocation;
+
+    /* ref pic list*/
+    if(avc_state->ref_pic_select_list_supported)
+    {
+        width = ALIGN(frame_width_in_mbs * 8,64);
+        height= frame_height_in_mbs ;
+        allocate_flag = i965_gpe_allocate_2d_resource(i965->intel.bufmgr,
+                                     &avc_surface->res_ref_pic_select_surface,
+                                     width, height,
+                                     width,
+                                     "Ref pic select list buffer");
+        if (!allocate_flag)
+            goto failed_allocation;
+    }
+
+    /*direct mv*/
+    avc_surface->dmv_top =
+        dri_bo_alloc(i965->intel.bufmgr,
+        "direct mv top Buffer",
+        68 * frame_mb_nums,
+        64);
+    avc_surface->dmv_bottom =
+        dri_bo_alloc(i965->intel.bufmgr,
+        "direct mv bottom Buffer",
+        68 * frame_mb_nums,
+        64);
+    assert(avc_surface->dmv_top);
+    assert(avc_surface->dmv_bottom);
+
+    return VA_STATUS_SUCCESS;
+
+failed_allocation:
+    return VA_STATUS_ERROR_ALLOCATION_FAILED;
+}
+
+static VAStatus
+gen9_avc_allocate_resources(VADriverContextP ctx,
+                            struct encode_state *encode_state,
+                            struct intel_encoder_context *encoder_context)
+{
+    struct i965_driver_data *i965 = i965_driver_data(ctx);
+    struct encoder_vme_mfc_context * vme_context = (struct encoder_vme_mfc_context *)encoder_context->vme_context;
+    struct gen9_avc_encoder_context * avc_ctx = (struct gen9_avc_encoder_context * )vme_context->private_enc_ctx;
+    struct generic_enc_codec_state * generic_state = (struct generic_enc_codec_state * )vme_context->generic_enc_state;
+    struct avc_enc_state * avc_state = (struct avc_enc_state * )vme_context->private_enc_state;
+    unsigned int size  = 0;
+    unsigned int width  = 0;
+    unsigned int height  = 0;
+    unsigned char * data  = NULL;
+    int allocate_flag = 1;
+    int i = 0;
+
+    /*all the surface/buffer are allocated here*/
+
+    /*second level batch buffer for image state write when cqp etc*/
+    i965_free_gpe_resource(&avc_ctx->res_image_state_batch_buffer_2nd_level);
+    size = INTEL_AVC_IMAGE_STATE_CMD_SIZE ;
+    allocate_flag = i965_allocate_gpe_resource(i965->intel.bufmgr,
+                             &avc_ctx->res_image_state_batch_buffer_2nd_level,
+                             ALIGN(size,0x1000),
+                             "second levle batch (image state write) buffer");
+    if (!allocate_flag)
+        goto failed_allocation;
+
+    /* scaling related surface   */
+    if(avc_state->mb_status_supported)
+    {
```

Comment:
minor suggestion. keep the identical coding type as others. like this
if () {

}